src/Memory.cpp

```diff
@@ -0,0 +1,74 @@
+/** \file  Memory.h
+ *  \brief Provide memory routines that terminate on failure
+ */
+
+#include "Memory.h"
+
+#include <cstdlib>
+#include <cstring>
+#include <cerrno>
+
+#include "Error.h"
+
+void* Memory::xmalloc(std::size_t size) noexcept
+{
+  /* Ensure we're going to allocate some memory.  */
+  if (size == 0)
+  {
+    size = 1;
```

Comment:
I wonder if a warning should be printed to the log if not an error and exit. I can't off the top of my head think of a valid reason to request an allocation of 0 bytes. If that's the case, it would help to discover logic errors in the code rather than create a 1-byte heap allocation. My two cents at least.

Reply:
This is not unusual as we've had instances where we're allocating `0` elements of something (validly) and the difference in platform behaviours (as this implementation defined behaviour) caused issues.

I do agree a logged comment would be useful, but that we then allow it to pass.